imagetest/test_suites/image_validation/image_validation_test.go

```diff
@@ -0,0 +1,165 @@
+package imagevalidation
+
+import (
+	"crypto/md5"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+)
+
+var (
+	runtest = flag.Bool("runtest", false, "really run the test")
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if *runtest {
+		os.Exit(m.Run())
+	} else {
+		os.Exit(0)
+	}
+}
+
+func TestHostname(t *testing.T) {
+	metadataHostname, err := utils.GetMetadata("hostname")
+	if err != nil {
+		t.Fatalf(" still couldn't determine metadata hostname")
+	}
+
+	// 'hostname' in metadata is fully qualified domain name.
+	shortname := strings.Split(metadataHostname, ".")[0]
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		t.Fatalf("couldn't determine local hostname")
+	}
+
+	if hostname != shortname {
+		t.Errorf("hostname does not match metadata. Expected: %q got: %q", shortname, hostname)
+	}
+
+	// If hostname is FQDN then lots of tools (e.g. ssh-keygen) have issues
+	if strings.Contains(hostname, ".") {
+		t.Errorf("hostname contains '.'")
+	}
+}
+
+// TestCustomHostname tests the 'fully qualified domain name', using the logic in the `hostname` utility.
```

Comment:
Looks like you renamed this function and forgot to update the doc comment to match.